src/main/kotlin/com/hypto/iam/server/authProviders/BaseAuthProvider.kt

```diff
@@ -0,0 +1,8 @@
+package com.hypto.iam.server.authProviders
+
+import com.hypto.iam.server.security.OAuthUserPrincipal
+import com.hypto.iam.server.security.TokenCredential
+
+interface BaseAuthProvider {
```

Comment:
Add another method `getProviderName` which returns a string and upon initialization, register it into the `AuthProviderRegistry`. So that, individual auth providers are responsible for their names and registering themselves into the registry.

Reply:
Done, need to fix `Leaking 'this' in constructor of non-final class BaseAuthProvider `